modules/gapi/CMakeLists.txt

```diff
@@ -257,6 +257,10 @@ if(TARGET opencv_test_gapi)
   if (HAVE_NGRAPH)
       ocv_target_compile_definitions(opencv_test_gapi PRIVATE -DHAVE_NGRAPH)
   endif()
+  if(HAVE_D3D11 AND HAVE_OPENCL)
+      ocv_target_include_directories(opencv_test_gapi PRIVATE ${OPENCL_INCLUDE_DIRS})
+      ocv_target_link_libraries(opencv_test_gapi PRIVATE "${OPENCL_LIBRARIES}")
```

Comment:
OpenCL should not be used directly in OpenCV modules/tests. OpenCV with OpenCL support is build without `OPENCL_LIBRARIES` definition by default.

They must use wrappers from the `core` module.

---

Only samples are allowed to use `find_package(OpenCL)`.

Reply:
Does it mean that I can't use `clFunctions` in tests? `clEnqueueReadImage` for example.
I can't include "opencv2/core/opencl/runtime/opencl_core.hpp". CI failed (CL/cl.h: No such file or directory).
And I can't use `enqueueReadImage` wrapper, I mustn't use cl.hpp.

Reply:
Currently we don't have tests which uses RAW cl* functions in tests.
OpenCV tests test OpenCV wrappers for OpenCL, not OpenCL functions.

OpenCV is always compiled with OpenCL, but it loads it dynamically.
Need to do NOT break this design even in tests.

---

Original comment is about `OPENCL_LIBRARIES`.

Reply:
Removed test with clFunction